File: front/src/types/notification.ts
import type { EntityId, ISODateString, UserSummary } from "@/types/community";

export type NotificationType = "post_like" | "comment" | "reply" | "follow";

export interface NotificationItem {
  id: EntityId;
  type: NotificationType;
  actor: UserSummary;
  postId: EntityId | null;
  commentId: EntityId | null;
  readAt: ISODateString | null;
  createdAt: ISODateString;
  updatedAt: ISODateString;
}

File: backend/app/schemas/notification.py
"""通知 API schema。"""

from __future__ import annotations

from datetime import datetime

from app.schemas.common import BaseSchema, ORMSchema
from app.schemas.community import UserSummarySchema


class NotificationSchema(ORMSchema):
    id: str
    type: str
    actor: UserSummarySchema
    post_id: str | None = None
    comment_id: str | None = None
    read_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class UnreadCountSchema(BaseSchema):
    count: int


class MarkAllReadSchema(BaseSchema):
    updated: int
